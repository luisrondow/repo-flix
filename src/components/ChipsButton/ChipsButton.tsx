import { ChipsButtonProps } from './ChipsButton.types'

const ChipsButton = (props: ChipsButtonProps) => {
  const { children, isSelected, onClick } = props

  return (
    <div
      data-testid="chips-button"
      onClick={onClick}
      className={`${
        isSelected ? 'bg-gray-500' : 'bg-white'
      } mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0`}
    >
      <div className="max-w-full flex-initial text-lg font-bold leading-none">{children}</div>
    </div>
  )
}

export default ChipsButton
