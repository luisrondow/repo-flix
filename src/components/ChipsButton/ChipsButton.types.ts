export type ChipsButtonProps = {
  children: React.ReactNode
  onClick?: () => void
  isSelected?: boolean
}
