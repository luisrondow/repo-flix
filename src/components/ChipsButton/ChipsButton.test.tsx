import { act, render } from '@testing-library/react'
import ChipsButton from './ChipsButton'

describe('ChipsButton', () => {
  const onClick = jest.fn()

  it('should render successfully', () => {
    const { container, getByText } = render(<ChipsButton onClick={onClick}>Test</ChipsButton>)

    const button = getByText('Test')

    expect(button).toBeInTheDocument()

    act(() => button.click())

    expect(onClick).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })

  it('should render selected', () => {
    const { container, getByTestId } = render(
      <ChipsButton isSelected onClick={onClick}>
        Test
      </ChipsButton>,
    )

    expect(getByTestId('chips-button')).toHaveClass('bg-gray-500')
    expect(container).toMatchSnapshot()
  })
})
