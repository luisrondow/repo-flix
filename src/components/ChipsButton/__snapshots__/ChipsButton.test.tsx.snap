// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipsButton should render selected 1`] = `
<div>
  <div
    class="bg-gray-500 mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
    data-testid="chips-button"
  >
    <div
      class="max-w-full flex-initial text-lg font-bold leading-none"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`ChipsButton should render successfully 1`] = `
<div>
  <div
    class="bg-white mb-4 flex cursor-pointer items-center justify-center rounded-full border border-black py-3 px-8 font-medium text-black sm:mb-0"
    data-testid="chips-button"
  >
    <div
      class="max-w-full flex-initial text-lg font-bold leading-none"
    >
      Test
    </div>
  </div>
</div>
`;
